bindgen/clang.rs

```diff
@@ -1524,6 +1564,42 @@ impl SourceLocation {
     }
 }
 
+impl SourceLocation {
+    /// Compare source locations, also considering `#include` directives.
+    pub(crate) fn partial_cmp_with_context(
+        &self,
+        other: &Self,
+        ctx: &BindgenContext,
+    ) -> Option<cmp::Ordering> {
```

Comment:
If I understood correctly, the only case where this returns `None` is if neither the file is included in the other. Given that this is only called inside `visit_sorted` wouldn't it be clearer to just return `cmp::Ordering::Equal` in that case?